raft/server.cc

```diff
@@ -495,13 +510,13 @@ future<add_entry_reply> server_impl::execute_add_entry(server_id from, command c
     if (from != _id && !_fsm->get_configuration().contains(from)) {
         // Do not accept entries from servers removed from the
         // configuration.
-        co_return add_entry_reply{not_a_leader{server_id{}}};
+        co_return add_entry_reply{transient_error{format("Server {} is not part of the configuration", from)}};
```

Comment:
Why are we throwing a "retryable" error here? Can this situation somehow be automatically resolved?

Reply:
Before the patch we threw `not_a_leader` which means that the node has to wait for a leader before retrying and it will never retry because it is outside of the config and will never see a leader again.